src/pipelines/worldpop/worldpop_population.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import zipfile
+from io import BytesIO
+from functools import partial
+from typing import Any, Dict, List, Tuple
+
+import requests
+from tqdm import tqdm
+from pandas import DataFrame, Series, read_csv, isnull
+
+from lib.concurrent import thread_map
+from lib.net import download
+from lib.pipeline import DataSource
+from lib.cast import age_group
+
+
+class WorldPopPopulationDataSource(DataSource):
+    """ Retrieves the requested properties from Wikidata for all items in metadata.csv """
+
+    url_strat_tpl = (
+        "ftp://ftp.worldpop.org.uk/GIS"
+        "/AgeSex_structures/Global_2000_2020/2020/"
+        "{code}/{code_lower}_{sex}_{age_bucket}_2020.tif"
+    )
+
+    url_total_tpl = (
+        "ftp://ftp.worldpop.org.uk/GIS"
+        "/Population/Global_2000_2020/2020/{code}/{code_lower}_ppp_2020.tif"
+    )
+
+    @staticmethod
+    def aggregate_population(data: Series) -> Dict[str, int]:
+        age_bucket_size = 10
+        age_bucket_count = 10
+        age_bucket_pairs = [
+            (i * age_bucket_size, (i + 1) * age_bucket_size - 1) for i in range(age_bucket_count)
+        ]
+
+        agg = {
+            "key": data["key"],
+            "population": 0,
+            "population_male": 0,
+            "population_female": 0,
+            **{f"population_age_{lo:02d}_{hi:02d}": 0 for lo, hi in age_bucket_pairs},
```

Comment:
Wouldn't this formatting print 00-09 instead of 0-9? Is that the correct output?

Reply:
Yes, that's the intent :-)

Reply:
But cast.age_group called in line 91 returns 0-9, which is probably why the 00-09 buckets in demographics.csv are population zero.

Reply:
Oh, you are right! I fixed this in a bit of a hacky way... The original intent was to reuse `cast.age_group` since the logic is 99% identical. I re-wrote this in a slightly more robust way.